### PR TITLE
feat: add proxy pallet

### PR DESCRIPTION
**Pull request by Roznovjak**
Fixes: #286

### CODE REVIEW

**Comment by Roznovjak** [comment]
@jak-pan What extrinsics do we want to allow for proxy accounts?

File: runtime/common/src/lib.rs [hunk 1]
```diff
@@ -111,6 +113,27 @@ parameter_types! {
...
+	Any,
+}
+impl Default for ProxyType {
+	fn default() -> Self {
+		Self::Any
+	}
+}
+
+parameter_types! {
+	pub ProxyDepositBase: Balance = 4 * DOLLARS + 480 * MILLICENTS;
```
Comment (vgantchev) [inline runtime/common/src/lib.rs:129]: is there some logic behind these amounts?

Comment (Roznovjak) [reply runtime/common/src/lib.rs:129]: This value should be equivalent to what Acala uses.

Comment (vgantchev) [reply runtime/common/src/lib.rs:129]: equivalent, meaning that it takes into account that Snek supply is 100x Acala?

Comment (Roznovjak) [reply runtime/common/src/lib.rs:129]: I used expected dollar values. https://github.com/AcalaNetwork/Acala/blob/ec911c5db90aef7cce00d3143415886369860216/runtime/acala/src/constants.rs#L37

Comment (jak-pan) [reply runtime/common/src/lib.rs:129]: > equivalent, meaning that it takes into account that Snek supply is 100x Acala?

yes dollars mean $ dollar value in sneks

**Review by lumir-mrkva — CHANGES_REQUESTED** [review]
proxy filter has to be defined properly

examples:
kusama - https://github.com/paritytech/polkadot/blob/master/runtime/kusama/src/lib.rs#L1084
polkadot - https://github.com/paritytech/polkadot/blob/master/runtime/polkadot/src/lib.rs#L1082
karura - https://github.com/AcalaNetwork/Acala/blob/master/runtime/karura/src/lib.rs#L1245
acala - https://github.com/AcalaNetwork/Acala/blob/master/runtime/acala/src/lib.rs#L1237

**Comment by Roznovjak** [comment]
not sure if we also want to include `NonTransfer` proxy group or some other use case specific groups.

**Comment by lumir-mrkva** [comment]
add transfer group

**Comment by Roznovjak** [comment]
I have decided to not include cross-chain transfers in the transfer proxy group.

**Review by lumir-mrkva — APPROVED** [review]

**Pull request merged**